### PR TITLE
Submit button placed in Questionnaire activity top app bar

### PR DESCRIPTION
**Pull request by kashyapj2793**
#131

### CODE REVIEW

**Comment by jingtang10** [comment]
@kashyapj2793 can you attach a screenshot here for @fredhersch to comment on? thanks!

File: datacapturegallery/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -16,4 +16,5 @@
 
 <resources>
     <string name="app_name">Data Capture Gallery</string>
+    <string name="submit">Submit</string>
```
Comment (jingtang10) [inline datacapturegallery/src/main/res/values/strings.xml:19]:
```suggestion
    <string name="submit_button_text">Submit</string>
```

File: datacapture/src/main/res/layout/questionnaire_fragment.xml [hunk 2]
```diff
@@ -28,10 +28,4 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             app:layoutManager="LinearLayoutManager"/>
-
-    <Button
-            android:id="@+id/submit"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:text="Submit"/>
 </LinearLayout>
```
Comment (jingtang10) [inline datacapture/src/main/res/layout/questionnaire_fragment.xml:31]: Can you try to get rid of this linear layout altogether please? Just to simplify the layout a bit.

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt [hunk 3]
```diff
@@ -37,31 +40,43 @@ class QuestionnaireActivity : AppCompatActivity() {
...
     }
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.top_bar_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val id: Int = item.getItemId()
+        return if (id == R.id.action_submit) {
+            questionnaireFragment.returnQuestionnaireResult()
```
Comment (jingtang10) [inline datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt:76]: use support fragmennt manager to retrieve the questionnaire fragment.

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt [hunk 4]
```diff
@@ -25,6 +27,7 @@ import com.google.android.fhir.datacapture.QuestionnaireFragment
 
 class QuestionnaireActivity : AppCompatActivity() {
     private val viewModel: QuestionnaireViewModel by viewModels()
+    private lateinit var questionnaireFragment: QuestionnaireFragment
```
Comment (jingtang10) [inline datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt:30]: I think we can do without this. please see the comment below.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt [hunk 5]
```diff
@@ -41,15 +40,16 @@ class QuestionnaireFragment : Fragment() {
...
-        view.findViewById<Button>(R.id.submit).setOnClickListener {
-            val serializedResponse = FhirContext.forR4().newJsonParser()
-                .encodeResourceToString(viewModel.questionnaireResponse)
-            setFragmentResult(
-                QUESTIONNAIRE_RESPONSE_REQUEST_KEY,
-                bundleOf(QUESTIONNAIRE_RESPONSE_BUNDLE_KEY to serializedResponse)
-            )
-        }
+    // Returns the current questionnaire response to parent
+    fun returnQuestionnaireResult() {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt:46]:
```suggestion
    fun getQuestionnaireResponse(): QuestionnaireResponse {
```

just return the response directly. we can stop using the results API now if the submit action is initiated outside of the fragment.

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt [hunk 6]
```diff
@@ -37,31 +40,43 @@ class QuestionnaireActivity : AppCompatActivity() {
...
+                    QuestionnaireResponseDialogFragment.BUNDLE_KEY_CONTENTS to
+                      result.getString(
+                        QuestionnaireFragment.QUESTIONNAIRE_RESPONSE_BUNDLE_KEY)
+                  )
+                  dialogFragment.show(
+                    supportFragmentManager,
+                    QuestionnaireResponseDialogFragment.TAG
+                  )
+              }
             )
```
Comment (jingtang10) [inline datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt:62]: extract the logic to populate a dialog to a separate function which is called from inside the `onOptionsItemSelected` function.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt (hunk 6 above)
Comment (kashyapj2793) [reply datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt]: done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt (hunk 5 above)
Comment (kashyapj2793) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt]: done

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt (hunk 3 above)
Comment (kashyapj2793) [reply datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt]: done

File: datacapture/src/main/res/layout/questionnaire_fragment.xml (hunk 2 above)
Comment (kashyapj2793) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml]: done

**Comment by kashyapj2793** [comment]
> 
> 
> @kashyapj2793 can you attach a screenshot here for @fredhersch to comment on? thanks!

I have added a  screenshot to issue #131 so that all the images are in one place

File: datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt [hunk 7]
```diff
@@ -39,32 +43,46 @@ class QuestionnaireActivity : AppCompatActivity() {
...
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val id: Int = item.getItemId()
+        return if (id == R.id.action_submit) {
+            val questionnaireResponse = (supportFragmentManager.findFragmentByTag
+            (QUESTIONNAIRE_FRAGMENT_TAG) as QuestionnaireFragment).getQuestionnaireResponse()
+            displayQuestionnaireResponse(questionnaireResponse)
+            true
+        } else super.onOptionsItemSelected(item)
```
Comment (jingtang10) [inline datacapturegallery/src/main/java/com/google/android/fhir/datacapture/gallery/QuestionnaireActivity.kt:66]:
```suggestion
        return when (item.getItemId()) {
            R.id.action_submit -> {
                val questionnaireFragment = supportFragmentManager.findFragmentByTag(
                    QUESTIONNAIRE_FRAGMENT_TAG
                ) as QuestionnaireFragment
                displayQuestionnaireResponse(questionnaireFragment.getQuestionnaireResponse())
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
```

**Review by jingtang10 — APPROVED** [review]
this is great! thanks kashyap!

just one suggestion below.

**Pull request merged**